### PR TITLE
NO-JIRA: e2e:mixedcpus: check env variables of init process

### PR DESCRIPTION
**Pull request by Tal-or**
We had a bug (https://github.com/cri-o/cri-o/pull/7710) in CRI-O on which the Mixed-CPUs environment variables were injected to all processes but the init process.

The test has been modified to verify that the env variables are injected to the init process as well. From this point all child processes would inherit the Mixed-CPUs environment variables, unless removed explicitly, but we double check the child process anyway.

### CODE REVIEW

File: test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go [hunk 1]
```diff
@@ -457,3 +470,24 @@ func mustParse(cpus string) *cpuset.CPUSet {
 	Expect(err).ToNot(HaveOccurred(), "failed to parse cpuset; cpus=%q", cpus)
 	return &set
 }
+
+func fetchSharedCPUsFromEnv(c *kubernetes.Clientset, p *corev1.Pod, containerName string) *cpuset.CPUSet {
+	GinkgoHelper()
+	// we should check the environment variable under the init process (pid 1)
+	cmd := []string{
+		"/bin/cat",
+		"/proc/1/environ",
```
Comment (ffromani) [inline test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go:479]: nice idea. I was thinking about changing the base image to dump the env, but this is even better

**Review by ffromani — COMMENTED** [review]
/lgtm

very nice idea

**Comment by ffromani** [comment]
for my understanding, why we moved away from `BeEmpty()` ?

**Comment by Tal-or** [comment]
> for my understanding, why we moved away from `BeEmpty()` ?

The variables we checked were `cpuset.CPU` struct. you cannot check `BeEmpty()` for struct (only for string, array, map, chan, or slice)
It was done by me hurry up :) 
My actually meaning was to use `IsEmpty()` implemented by the   `cpuset.CPU` struct

**Comment by ffromani** [comment]
thanks, makes sense to me.
I would still add a better explanation around here: 

```
Expect(shared.IsEmpty()).ToNot(BeTrue())
```
because this is likely to generate not-so-helpful error message. I'd use something like

```
Expect(shared.IsEmpty()).ToNot(BeTrue(), "shared cpuset is empty")
```

**Comment by Tal-or** [comment]
> thanks, makes sense to me. I would still add a better explanation around here:
> 
> ```
> Expect(shared.IsEmpty()).ToNot(BeTrue())
> ```
> 
> because this is likely to generate not-so-helpful error message. I'd use something like
> 
> ```
> Expect(shared.IsEmpty()).ToNot(BeTrue(), "shared cpuset is empty")
> ```

Np will update in the close hour

**Comment by yanirq** [comment]
/label acknowledge-critical-fixes-only label
very low risk - not affecting payload

**Comment by yanirq** [comment]
/label acknowledge-critical-fixes-only

very low risk - not affecting payload

**Pull request merged**